Setup CI build on Windows + Bazel.

We need a build for Windows to verify the code compiles and runs there. The build should:

- [x] Compile the code (duh).
- [x] Run the unit tests.
- [ ] Run the integration tests.

Trying to build with bazel with the fix in google-cloud-cpp and getting these errors.

```
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(156): error C2559: 'std::tuple<std::tuple_element<I,std::tuple<_Types...>>::type...> google::cloud::spanner::v0::Row<Types...>::get(void) const': cannot overload a member function without ref-qualifier with a member function with ref-qualifier
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(256): note: see reference to class template instantiation 'google::cloud::spanner::v0::Row<Types...>' being compiled
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(205): error C2975: '_Size': invalid template argument for 'std::array', expected compile-time constant expression
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\array(20): note: see declaration of '_Size'
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(211): error C2975: '_Size': invalid template argument for 'std::array', expected compile-time constant expression
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\array(20): note: see declaration of '_Size'
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(247): error C2975: '_Size': invalid template argument for 'std::array', expected compile-time constant expression
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\array(20): note: see declaration of '_Size'
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(251): note: see reference to class template instantiation 'google::cloud::spanner::v0::Row<Types...>::InsertValue' being compiled
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(156): error C2559: 'std::tuple<std::tuple_element<I,std::tuple<>>::type...> google::cloud::spanner::v0::Row<>::get(void) const': cannot overload a member function without ref-qualifier with a member function with ref-qualifier
google/cloud/spanner/row_test.cc(61): note: see reference to class template instantiation 'google::cloud::spanner::v0::Row<>' being compiled
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(156): error C2559: 'std::tuple<std::tuple_element<I,std::tuple<bool>>::type...> google::cloud::spanner::v0::Row<bool>::get(void) const': cannot overload a member function without ref-qualifier with a member function with ref-qualifier
google/cloud/spanner/row_test.cc(67): note: see reference to class template instantiation 'google::cloud::spanner::v0::Row<bool>' being compiled
C:\t\4rkbi76t\execroot\com_github_googleapis_google_cloud_cpp_spanner\google/cloud/spanner/row.h(156): error C2559: 'std::tuple<std::tuple_element<I,std::tuple<bool,int64_t>>::type...> google::cloud::spanner::v0::Row<bool,int64_t>::get(void) const': cannot overload a member function without ref-qualifier with a member function with ref-qualifier
```

With visual studio 2019, the 2nd issue is gone. I see only 1 build failure for a test:

```
ERROR: T:/src/github/google-cloud-cpp-spanner/google/cloud/spanner/BUILD:71:2: Couldn't build file google/cloud/spanner/_objs/spanner_client_value_test/value_test.obj: C++ compilation of rule '//google/cloud/spanner:spanner_client_value_test' failed (Exit 2)
google/cloud/spanner/value_test.cc(126): error C3535: cannot deduce type for 'auto' from 'initializer list'
google/cloud/spanner/value_test.cc(126): error C3535: cannot deduce type for 'auto &&' from 'initializer list'
google/cloud/spanner/value_test.cc(127): error C2398: Element '1': conversion from '__int64' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
google/cloud/spanner/value_test.cc(129): error C2398: Element '3': conversion from 'unsigned long' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
google/cloud/spanner/value_test.cc(133): error C2398: Element '9': conversion from 'unsigned long' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
google/cloud/spanner/value_test.cc(134): error C2398: Element '10': conversion from '__int64' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
google/cloud/spanner/value_test.cc(477): error C3535: cannot deduce type for 'auto' from 'initializer list'
google/cloud/spanner/value_test.cc(477): error C3535: cannot deduce type for 'auto &&' from 'initializer list'
google/cloud/spanner/value_test.cc(478): error C2398: Element '1': conversion from '__int64' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
google/cloud/spanner/value_test.cc(480): error C2398: Element '3': conversion from 'unsigned long' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
google/cloud/spanner/value_test.cc(484): error C2398: Element '9': conversion from 'unsigned long' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
google/cloud/spanner/value_test.cc(485): error C2398: Element '10': conversion from '__int64' to '_Ty' requires a narrowing conversion
        with
        [
            _Ty=int
        ]
```

--- a/ci/kokoro/windows/bazel/build.bat
+++ b/ci/kokoro/windows/bazel/build.bat
@@ -15,6 +15,8 @@ REM limitations under the License.
 echo %date% %time%
 cd github\google-cloud-cpp-spanner
 
+set TMP=T:\tmp
+
 echo "Create the bazel output directory."
 echo %date% %time%
 if not exist "C:\b\" mkdir C:\b

--- a/ci/kokoro/windows/bazel/build.bat
+++ b/ci/kokoro/windows/bazel/build.bat
@@ -44,6 +44,28 @@ bazel --output_user_root=C:\b test ^
 
 if %errorlevel% neq 0 exit /b %errorlevel%
 
+echo "Running integration tests"
+call "%KOKORO_GFILE_DIR%/spanner-integration-tests-config.bat"
+
+@rem It seems like redirecting to a file is the easiest way to store the
+@rem command output to a variable.
+bazel --output_user_root=C:\b info output_base > t:\bazel-info.txt
+set /p BAZEL_OUTPUT_DIR=<t:\bazel-info.txt
+del t:\bazel-info.txt
+
+echo %date% %time%
+bazel --output_user_root=C:\b test ^
+  --keep_going ^
+  --test_output=errors ^
+  --verbose_failures=true ^
+  --test_tag_filters=integration-tests ^
+  --test_env GOOGLE_APPLICATION_CREDENTIALS=%KOKORO_GFILE_DIR%/spanner-credentials.json ^
+  --test_env GOOGLE_CLOUD_PROJECT=%GOOGLE_CLOUD_PROJECT% ^
+  --test_env GOOGLE_CLOUD_CPP_SPANNER_INSTANCE=%GOOGLE_CLOUD_CPP_SPANNER_INSTANCE% ^
+  --test_env GOOGLE_CLOUD_CPP_AUTO_RUN_EXAMPLES=yes ^
+  --test_env GRPC_DEFAULT_SSL_ROOTS_FILE_PATH=%BAZEL_OUTPUT_DIR%/external/com_github_grpc_grpc/etc/roots.pem ^
+  -- //google/cloud/spanner/...:all
+
 @echo %date% %time%
 @echo DONE DONE DONE "============================================="
 @echo DONE DONE DONE "============================================="

--- a/ci/kokoro/windows/bazel/build.bat
+++ b/ci/kokoro/windows/bazel/build.bat
@@ -66,6 +66,8 @@ bazel --output_user_root=C:\b test ^
   --test_env GRPC_DEFAULT_SSL_ROOTS_FILE_PATH=%BAZEL_OUTPUT_DIR%/external/com_github_grpc_grpc/etc/roots.pem ^
   -- //google/cloud/spanner/...:all
 
+if %errorlevel% neq 0 exit /b %errorlevel%
+
 @echo %date% %time%
 @echo DONE DONE DONE "============================================="
 @echo DONE DONE DONE "============================================="

--- a/ci/kokoro/windows/bazel/common.cfg
+++ b/ci/kokoro/windows/bazel/common.cfg
@@ -17,3 +17,4 @@ build_file: "google-cloud-cpp-spanner/ci/kokoro/windows/bazel/build.bat"
 timeout_mins: 120
 
 gfile_resources: "/bigstore/cloud-cpp-integration-secrets/spanner-credentials.json"
+gfile_resources: "/bigstore/cloud-cpp-integration-secrets/spanner-integration-tests-config.bat"

